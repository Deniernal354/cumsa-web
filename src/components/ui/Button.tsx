import type { ButtonHTMLAttributes, PropsWithChildren } from "react";

type Props = PropsWithChildren<{
  variant?: "primary" | "outline";
  className?: string;
}> &
  ButtonHTMLAttributes<HTMLButtonElement>;

export default function Button({ variant = "primary", className = "", children, ...props }: Props) {
  const base =
    "inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none";
  const styles =
    variant === "primary"
      ? "bg-[var(--blue)] text-white hover:opacity-90 focus-visible:ring-[var(--blue)] ring-offset-[var(--surface)] px-4 py-2"
      : "border border-black/10 bg-[var(--surface)] text-[var(--foreground)] hover:bg-black/5 focus-visible:ring-[var(--blue)] ring-offset-[var(--surface)] px-4 py-2";
  return (
    <button className={`${base} ${styles} ${className}`} {...props}>
      {children}
    </button>
  );
}
