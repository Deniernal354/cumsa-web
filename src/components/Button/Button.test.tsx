import { render } from "@testing-library/react";
import Button from "./Button";

test("Button snapshot", () => {
  const { asFragment } = render(<Button />);
  expect(asFragment()).toMatchSnapshot();
});
