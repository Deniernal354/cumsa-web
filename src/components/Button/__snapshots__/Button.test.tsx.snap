// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-[var(--blue)] text-white hover:opacity-90 focus-visible:ring-[var(--blue)] ring-offset-[var(--surface)] px-4 py-2 "
  />
</DocumentFragment>
`;
